File: java/src/main/java/deltix/luminary/LiteralFloat64.java
package deltix.luminary;

public final class LiteralFloat64 extends Literal {
    private final double value;

    public LiteralFloat64(double value) {
        super(LiteralKind.FLOAT64);
        this.value = value;
    }

    public double getValue() {
        return value;
    }

    @Override
    public String toString() {
        if (Double.isNaN(value))
            return "Float64.NaN";
        if (Double.isInfinite(value))
            return value > 0 ? "Float64.POSITIVE_INFINITY" : "Float64.NEGATIVE_INFINITY";
        return Double.toString(value) + "f64";
    }
}

File: csharp/main/LiteralFloat64.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralFloat64 : Literal, ILiteral<Double>
	{
		public LiteralFloat64(Double value)
			: base(LiteralKind.Float64)
		{
			Value = value;
		}

		public Double Value { get; }

		public override String ToString()
		{
			if (Double.IsNaN(Value))
				return "Float64.NaN";
			if (Double.IsPositiveInfinity(Value))
				return "Float64.POSITIVE_INFINITY";
			if (Double.IsNegativeInfinity(Value))
				return "Float64.NEGATIVE_INFINITY";
			return $"{Value:G17}f64";
		}
	}
}
